mne/epochs.py

```diff
@@ -818,9 +818,14 @@ def _reject_setup(self, reject, flat):
         # check for invalid values
         for rej, kind in zip((reject, flat), ("Rejection", "Flat")):
             for key, val in rej.items():
-                if val is None or val < 0:
+                if callable(val):
+                    continue
+                elif val is not None and val >= 0:
+                    continue
+                else:
                     raise ValueError(
-                        '%s value must be a number >= 0, not "%s"' % (kind, val)
+                        '%s value must be a number >= 0 or a valid function, not "%s"'
+                        % (kind, val)
                     )
 
         # now check to see if our rejection and flat are getting more
```

Comment:
One more thing, the `_reject_setup` function in epochs.py includes a section to "make sure new thresholds are at least as stringent as the old ones. I think this is used if you didn't preload and you run `drop_bad()`. The current implementation just checks to see if `reject[key] > old_reject[key]`. 

This doesn't work with callables, maybe I could try checking to see if the new reject parameters would rejects less epochs than the old ones, but I would probably have to loop through all the epochs and that seems unnecessarily complex so maybe I should just not allow changing reject and flat if either is a callable.

Reply:
>  I am not really sure why this conditional is necessary.

If you reject once with strict parameters A, and then ask to reject with more lenient parameters B it's a no-op and thus almost certainly an error. Users might do this thinking they'll "get back" the old ones we dropped, but once they're dropped they're gone forever.

> (My current implementation just skips the check for callables)

I think this is fine, users doing custom stuff will hopefully know once epochs are dropped they're gone.